### ISSUE TITLE
Do the ARIA TABS 1.2 fail WCAG 1.4.10 Reflow?

### ISSUE THREAD

**Issue by LaurenceRLewis** [w3c/aria-practices#2284]
When using the zoom instructions from the Understanding Success Criterion [1.4.10: Reflow](https://www.w3.org/WAI/WCAG21/Understanding/reflow.html) setting a screen width of 1280px @400% browser zoom the Tabpanel content is truncated and horizontal scrolling occurs.

My personal testing demonstrates that the Tabs do reflow or stack, however the Tabpanel content does not reflow.

ALT=ARIA Tabs example zoomed and content is cut off.
![Tabs-Only-zoomed](https://user-images.githubusercontent.com/4565139/162103478-ff908a9e-c986-4aec-8ca9-4c5edd822895.png)

ALT=Showing Chrome Developer tools where I have removed all other page content so that only the Tab widget is available.
![Tabs-Only-Code](https://user-images.githubusercontent.com/4565139/162103463-c8744f02-737d-443b-9b78-7117ce2bc7c3.png)

**Comment by mcking65** [comment]
Per conversation in August 2, 2022 Telecon:

* Since this example is intended to be the simplest possible implementation, reduce to 3 tabs and change minwidth from 550 to approximately 320 (potentially less depending on padding.
* This will make the current examples WCAG compliant.
* * We need a more advanced  tabs example that illustrates how to be WCAG compliant with more tabs. This could include a menu for choosing which tabs are displayed or chevrons that intelligently scroll the tablist.

**Comment by curtbellew** [comment]
> Per conversation in August 2, 2022 Telecon:
> 
> * Since this example is intended to be the simplest possible implementation, reduce to 3 tabs and change minwidth from 550 to approximately 320 (potentially less depending on padding.
> * This will make the current examples WCAG compliant.
> * * We need a more advanced  tabs example that illustrates how to be WCAG compliant with more tabs. This could include a menu for choosing which tabs are displayed or chevrons that intelligently scroll the tablist.

In taking a closer look at this, it looks like there will still be two scroll bars if we reduce to 3 tabs and lower the minwidth.  We would need to reduce to probably 2 tabs to prevent wrapping of the tabs and prevent a scroll bar at the bottom of the browser.    The intent of the standard is that there should be either a horizontal or vertical scroll bar at the browser page level, not both.  One or the other and they definitely prefer vertical scrolling.  
> The intent of this Success Criterion is to support people with low vision who need to enlarge text and read it in a single column. When the browser zoom is used to scale content to 400%, it reflows - i.e., it is presented in one column so that scrolling in more than one direction is not necessary.

There are also exceptions built in to this standard where the visual layout of the content is essential for the user to understand it's meaning.  Data tables are a prime example where reflowing the data table content would make it unusable visually.  This tabs example is the same sort of case in that wrapping the tabs causes the tab list to essentially lose it's meaning - the tabs no longer appear associated to one another or to their owned content.  

> Content which requires two-dimensional layout for usage or meaning cannot reflow without losing meaning, and is therefore out of scope of this Success Criterion. For example, graphics and video are by their nature two-dimensional. Cutting up an image and stacking the blocks would render the content unusable so that is out of scope. It is acceptable to provide two-dimensional scrolling for such parts of the content.

The allowance here is that we can provide two dimensional scrolling for the this content and not for the entire page.  We can achieve this via CSS: overflow-x:auto; on the outer div.
Following is a screen shot of the proposal, showing a vertical scrollbar at the browser level and a horizontal scroll bar a the content level.
![image](https://user-images.githubusercontent.com/45948345/215904798-71a8ea79-025b-49c8-8e42-fc8695c4a362.png)

**Comment by JAWS-test** [comment]
@curtbellew 

It is important that the content of the tabpanel does not have to be scrolled horizontally to read it!

If the tablist needs to be scrolled horizontally, I don't think it's that problematic, but I don't know if that meets any of the 1.4.10 exceptions.
For the tablist not to have to be scrolled horizontally, there are 3 possibilities:
* Reflow
* Show a button at the end of the visible tablist to display more tabs (via dropdown)
* Show a button on the right and left side of the visible tablist to show more tabs (a kind of scroll function without scrollbar)

**Comment by curtbellew** [comment]
> @curtbellew
> 
> It is important that the content of the tabpanel does not have to be scrolled horizontally to read it!
> 
> If the tablist needs to be scrolled horizontally, I don't think it's that problematic, but I don't know if that meets any of the 1.4.10 exceptions. For the tablist not to have to be scrolled horizontally, there are 3 possibilities:
> 
> * Reflow
> * Show a button at the end of the visible tablist to display more tabs (via dropdown)
> * Show a button on the right and left side of the visible tablist to show more tabs (a kind of scroll function without scrollbar)

These are some methods of addressing the issue but the point of this standard remains the 2 dimensional standard.  As with any standard there are many ways to address them.  In this particular case the most straightforward way to fix this issue is via the horizontal content scrolling as outlined in the standard.  That is outline within the standard, please feel free to review the standard.

**Comment by JAWS-test** [comment]
The current implementation of tabpanels is clearly a violation of SC 1.4.10 because they must be scrolled horizontally after each line of text.

The display has thus worsened significantly compared to the [initial findings](https://github.com/w3c/aria-practices/issues/2284#issuecomment-1091993806).
